### DISCUSSION TITLE
Revising Plots

### DISCUSSION THREAD

**Discussion by lucas-wilkins** [SasView/sasview#2475] [Ideas]
I just want to kick of a discussion about plotting. Big topics that need thinking about...

1) Its clear that a one-size fits all approach is reaching its limits, what class structures do we want going forward.
  a) What are the individual types of plots that we want?
  b) How do we organise these types to make best use of inheritance?
2) There's definitely too many plots floating about, a solution to this requires answering the following:
  a) How are we going to group plots together, possibilities include tabbed group windows and subplots?
  b) How are we going to link individual plots to data/calculations/etc?
  c) How flexible would we like to make any grouping?
  d) What are some feasible strategies for approaching this?

The following are some basic questions that will help answer the ones above, please adjust them if you have ideas...

What type of plots need to be supported?
* Data plots (1D and 2D)
* Model plots
* Polydispersity plots
* Convergence
* Corner plots
* P(r) plots
* 3D data structures and beyond

There are potentially some different behaviours for different kinds of data, things to consider are:
* Densities vs distributions
* Real numbers, positive reals, integers ...
* Appropriate scaling choices for different data
* Direction of axes
* Different formatting options on different kinds of plot
* Do we link units?

**Comment by lucas-wilkins** [comment]
Pinging @gonzalezma @butlerpd @krzywon @wpotrzebowski @rozyczko

**Comment by butlerpd** [comment]
Thanks for starting the discussion. Particularly timely if there might be somebody to work on the issue in the near future.  I think there are some issues (old ones?) that should probably be linked to the discussion (I'm sure Jeff will be all over that if they exist 😄)

This is however a huge space and I am thinking it may be important to figure out how to subdivide if we want to see real progress any time soon? that said, the problem, like many in SasView, is that everything is intertwined at some level ☹️  Still...

To now expand the scope even more (doing what I warn against of course), something I have been wondering for a while is whether we should not do a better job of integrating with matplotlib rather than having the tendency to rewrite its functionality ourselves which sometimes bites us. A recent example being the axis scaling and properties.  I suspect that the kludgy looking transforms code we have (for the scale function leading to Guinier, Kratky and other plots) could probably be done with the custom transforms in matplotlib?  Anyway - more food for thought
